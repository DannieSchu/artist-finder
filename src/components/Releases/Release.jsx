import React from 'react';
import PropTypes from 'prop-types';
import styles from '../Artists/Artist.jsx';

const Release = ({ title, date, url = 'music-icon.svg' }) => {
  return (
    <table className={styles.Artist}>
      <tr>
        <th>Release</th>
        <td>{title}</td>
      </tr>
      <tr>
        {date && <td>{date}</td>}
        {date && <th>Date</th>}
      </tr>
      <image src={url} alt={url ? name : 'Default image'} />
    </table>
  );
};

Release.propTypes = {
  title: PropTypes.string.isRequired,
  date: PropTypes.string,
  url: PropTypes.string
};

export default Release;

{/* <>
<tr>
  <th>Artist</th>
  <td>{name}</td>
</tr>
<tr>
  {origin && <th>Origin</th>}
  {origin && <td>{origin}</td>}
</tr>
<tr>
  {birth && <th>Birth</th>}
  {birth && <td>{birth.slice(0, 4)}</td>}
</tr>
<tr>
  {death && <th>Death</th>}
  {death && <td>{death.slice(0, 4)}</td>}
</tr>
</> */}

{/* <tr>
        <th>Artist</th>
      </tr>
      <tr>
        <td>{name}</td>
      </tr>
      <tr>
        {origin && <th>Origin</th>}
      </tr>
      <tr>
        {origin && <td>{origin}</td>}
      </tr>
      <tr>
        {birth && <th>Birth</th>}
        {death && <th>Death</th>}
      </tr>
      <tr>
        {birth && <td>{birth.slice(0, 4)}</td>}
        {death && <td>{death.slice(0, 4)}</td>}
      </tr> */}
