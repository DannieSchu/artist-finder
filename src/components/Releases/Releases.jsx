import PropTypes from 'prop-types';
import Release from './Release.jsx';
import withList from '../../utils/withList.jsx';

const Releases = withList(Release, 'releases');

Releases.propTypes = {
  releases: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string,
  })).isRequired
};

export default Releases;
