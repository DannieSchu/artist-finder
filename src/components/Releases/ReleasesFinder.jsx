import React from 'react';
import PropTypes from 'prop-types';
import Release from './Release.jsx';
import withPaging from '../../utils/withPaging.jsx';
import withList from '../../utils/withList.jsx';
import { useGetReleases } from '../../hooks/getReleases.jsx';

const ReleasesFinder = ({ match, page, setTotalPages }) => {
  const { releases, artist, loading } = useGetReleases(match, page, setTotalPages);
  const Releases = withList(Release, 'releases');

  return (
    <Releases 
      releases={releases} 
      heading={artist} 
      loading={loading} 
    />
  );
};

ReleasesFinder.propTypes = {
  page: PropTypes.number.isRequired,
  setTotalPages: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      artistId: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default withPaging(ReleasesFinder);
