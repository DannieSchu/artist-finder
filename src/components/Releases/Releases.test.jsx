import React from 'react';
import { shallow } from 'enzyme';
import Releases from './Releases.jsx';

describe('Releases', () => {
  it('matches a snapshot', () => {
    const releases = [
      { title: 'some title',
        id: 'fneowigbw0' },
      { title: 'another title', 
        id: 'fneowigbw0' },
      { title: 'a third title',
        id: 'fneowigbw0' }
    ];

    const wrapper = shallow(
      <Releases releases={releases} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
