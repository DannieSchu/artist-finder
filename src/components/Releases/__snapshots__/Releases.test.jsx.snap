// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Releases matches a snapshot 1`] = `
<section
  className="List"
>
  <h2>
    Results for '
    '
  </h2>
  <ul>
    <li
      key="fneowigbw0"
    >
      <Link
        to="/release/fneowigbw0"
      >
        <Release
          id="fneowigbw0"
          title="some title"
        />
      </Link>
    </li>
    <li
      key="fneowigbw0"
    >
      <Link
        to="/release/fneowigbw0"
      >
        <Release
          id="fneowigbw0"
          title="another title"
        />
      </Link>
    </li>
    <li
      key="fneowigbw0"
    >
      <Link
        to="/release/fneowigbw0"
      >
        <Release
          id="fneowigbw0"
          title="a third title"
        />
      </Link>
    </li>
  </ul>
</section>
`;
